docs/java/how-to-develop-a-workflow-definition-in-java.md

```diff
@@ -135,23 +227,187 @@ This registration fails with the following message:
+
+Related references:
+
+- [How to spawn a Workflow Execution in Java](/docs/java/how-to-spawn-a-workflow-execution-in-java).
+- `WorkflowStub.java` reference: <https://github.com/temporalio/sdk-java/blob/master/temporal-sdk/src/main/java/io/temporal/client/WorkflowStub.java>
+
+### Calling other Workflows
+
+Workflows can invoke, and send Signals to, other Workflows purely by name. This helps particularly for executing Workflows written in other language SDKs.
+
+See the [Temporal Polyglot example](https://github.com/tsurdilo/temporal-polyglot) for more information.
+
+### Dynamic Workflows
+
+Use `DynamicWorkflow` to implement Workflow Types dynamically. When you register a Workflow implementation type that extends `DynamicWorkflow`, it can be used to implement any Workflow Type that is not explicitly registered with the Worker.
+
+The main use case for `DynamicWorkflow` is an implementation of custom Domain Specific Languages (DSLs). A single implementation can implement a Workflow Type which by definition is dynamically loaded from some external source.
+You can also use `DynamicWorkflow` when you need a default Workflow that can handle all Workflow Types that are not registered with a Worker.
+
+The Dynamic Workflow interface is implemented with the `execute` method. This method takes in `EncodedValues` that are inputs to the Workflow Execution. These inputs can be specified by the Client when invoking the Workflow Execution.
+
+```java
+public class MyDynamicWorkflow implements DynamicWorkflow {
+   @Override
+    public Object execute(EncodedValues args) {
+    }
+}
+```
+
+The `DynamicWorkflow` must be registered with a Worker.
+The following rules apply when registering a Dynamic Workflow with a Worker:
+
+- Only one Workflow implementation that extends `DynamicWorkflow` can be registered with a Worker.
+- You can register multiple type-specific Workflow implementations alongside a single `DynamicWorkflow` implementation.
+- Implement the `execute` method for Dynamic Workflow implementations. Do not specify a `@WorkflowMethod` when using Dynamic Workflows.
+- Implement Signals and Queries dynamically.
+
+  Example for implementing Signal handler dynamically:
+
+  ```java
+        Workflow.registerListener(
+          (DynamicSignalHandler)
+              (signalName, encodedArgs) -> name = encodedArgs.get(0, String.class));
+  ```
+
+  Example for implementing Query handler dynamically:
+
+  ```java
+        Workflow.registerListener(
+          (DynamicQueryHandler)
+              (queryType, encodedArgs) -> {
+              return name;
+      });
+  ```
+
+  Note that `DynamicSignalHandler` and `DynamicQueryHandler` can also be implemented in regular Workflow implementations.
+
+- Since `DynamicWorkflow` can be invoked for different Workflow Types, to check what type is running when your Dynamic Workflow `execute` method runs, use:
```

Comment:
```suggestion
- Because `DynamicWorkflow` can be invoked for different Workflow Types, to check what type is running when your Dynamic Workflow `execute` method runs, use:
```